tern/analyze/docker/dockerfile.py

```diff
@@ -265,3 +265,38 @@ def get_base_image_tag(base_instructions):
     if len(image_tag_list) == 1:
         image_tag_list.append('')
     return tuple(image_tag_list)
+
+
+def find_git_info(line, dockerfile_path):
+    '''Given a line of ADD command and the absolute path of dockerfile,
+    return the information(string format) on the git project name and sha
+    if a folder or tar containing a git repository.
+    ADD command has a general format:
+    ADD [--chown=<user>:<group>] <src> <dst>
+    '''
+    args = line.split(' ')
+    src_path = ''
+    # check if --chown exists
+    if args[1].startswith('--chown'):
+        # check if the line is valid
+        if len(args) < 4:
+            raise ValueError('Invalid ADD command line')
+        src_path = args[2]
+    else:
+        # the line has no --chown option
+        if len(args) < 3:
+            raise ValueError('Invalid ADD command line')
+        src_path = args[1]
+    # get the absolute src_path
+    comment_line = general.check_git_src(dockerfile_path, src_path)
+    return comment_line
+
+
+def expand_add_command(dfobj):
+    dockerfile_path = dfobj.filepath
```

Comment:
I'm not sure if `dfobj.filepath` will necessarily be an absolute path. @nishakm is it possible that this path could be relative? From what I can tell, `dfobj.filepath` gets assigned to whatever the user specifies as the Dockerfile, which may not be absolute.

Reply:
I use open() and tarfile.open() to read the src file of ADD command. So once we make sure the relative path is in the sys.path, we can also get the src file.

Reply:
@rnjudge No it's not an absolute path and it doesn't need to be that. We just need to find the git repo and commit HEAD. I don't think we need to read what is in the tar file.